noxfile.py

```diff
@@ -4,10 +4,96 @@
 
 nox.options.default_venv_backend = "uv"
 
+supported_python_versions = ["3.10", "3.11", "3.12"]
 
-maxpython = "3.12"
+maxpython = max(supported_python_versions)
+minpython = min(supported_python_versions)
 
-doc_requirements = ("-r", "ci_requirements/requirements_docs_py312.txt")
+requirements_directory = "ci_requirements"
+
+
+def get_requirements_file(
+    category: str, version: str, resolution: str = "highest"
+) -> str:
+    """
+    Return the file path to the requirements file.
+
+    Parameters
+    ----------
+    category : str
+        The category for determining requirements: "docs", "tests", or
+        "all".
+
+    version : str
+        The version of Python to get the requirements for.
+
+    resolution : str
+        The resolution strategy to be used by ``uv pip compile``
+        ``uv pip compile``.
+
+    Returns
+    -------
+    str
+        The path to the requirements file.
+    """
+    specifiers = [category, version, resolution]
+    return f"{requirements_directory}/{'-'.join(specifiers)}.txt"
+
+
+@nox.session
+def requirements(session):
+    """Regenerate pinned requirements files used during CI."""
+
+    session.install("uv >= 0.1.37")
+
+    command = (
+        "python",
+        "-m",
+        "uv",
+        "pip",
+        "compile",
+        "pyproject.toml",
+        "--upgrade",
+        "--quiet",
+    )
```

Comment:
```suggestion
    command = "python -m uv pip compile pyproject.toml --upgrade --quiet"
    command = command.split()
```
🎨 I'm wondering if it would be more readable to do something like this. 🤔 There are tradeoffs, I suppose. ⚖️